### ISSUE TITLE
Genomics API fails to decode VariantInfo and VariantCallInfo

### ISSUE THREAD

**Issue by chriswhelix** [google/google-api-go-client#130]
Calls to the v1 Genomics API using this library fail to correctly decode the VariantInfo and VariantCallInfo fields, due to their type definition as empty structs. The correct type definition is a map[string][]string, as made quite clear by the comment. Changing this type causes the JSON to be decoded correctly:

https://github.com/chriswhelix/google-api-go-client/commit/0fc37fc6584d89ce2c537b6a6f422b72f35316d9

However, I don't know how to fix the generator to create the file with the correct type statements.

**Comment by mcgreevy** [comment]
It looks like the json specification of the API says that these fields are arrays of type "any" (which the generator does not handle), rather than arrays of type "string" (which would seem to be the appropriate type given the comment, and which the generator can handle).

I will dig into why the why the API is configured this way.

**Comment by calbach** [comment]
Responding for the Google Genomics team, I independently just noticed this issue.

In proto this field is `map<string, ListValue>`, for example [Variant.info](https://github.com/googlegenomics/utils-java/blob/master/src/main/proto/google/genomics/v1/variants.proto#L264). We would have liked to type this more strictly but doing so would not produce the desired JSON  given the current proto3 JSON specification. We're also a bit forced on the exact JSON output being `{"": [""]}` due to our compliance with the Global Alliance specification (http://github.com/ga4gh/schemas).

To expand a bit:
```
message StringList { repeated string strings = 1; }
message Foo { map<string, StringList> bar = 1; }
```
yields: `{bar: {"": {strings: [""]}}}`

In summary, the API is not changing for this version. Unfortunately these fields are pretty critical to our API, so this issue is definitely a blocker for clients using the Go library. I'm going to dig a bit as to why the JSON specification is showing "any" here - `ListValue` should be usable, albeit cumbersome.

**Comment by calbach** [comment]
OK, digging a bit more, "any" is the correct and expected specification and should not be confused with the proto3 `Any` type.

Is there an issue tracking "any" support? I may be able to volunteer some time to work on this, though I don't have a good sense of how large a project this would be.

**Comment by mcgreevy** [comment]
@chriswhelix , @calbach 

I think that my patch here:
  https://code-review.googlesource.com/4191
will fix this issue.

I'm not familiar with the genomics API, so it's difficult for me to test that it works as intended. Can you please try it out and let me know if it works for you?

**Comment by calbach** [comment]
Looks good, thanks. I patched this in, wrote some code to query the genomics API, and verified that I'm now able to unpack the `info` fields.

**Comment by chriswhelix** [comment]
Yep, this fix worked for me, as well. Thanks for the quick attention to this!
